src/audio/buffer.c

```diff
@@ -180,7 +180,7 @@ void buffer_free(struct comp_buffer *buffer)
 		.buffer = buffer,
 	};
 
-	if (!buffer)
+	if (!buffer || atomic_get(&buffer->ref_count))
```

Comment:
How do we guarantee buffer R/W positions are valid if buffer is shared by 2 streams and other multiples cores ?